### PR TITLE
Use a simpler delayed loop

### PR DESCRIPTION
**Pull request by ammongit**
Having seen the real-life performance and ramifications of the delayed event loop (as well as some issues where the loop would not function), I have decided to greatly simplify how the event loop operates.

### CODE REVIEW

**Comment by ammongit** [comment]
Thanks @Ma-wa-re

File: futaba/delayed.py [hunk 1]
```diff
@@ -40,20 +39,22 @@ def push(self, coro):
         self.queue.put_nowait(coro)
 
     async def main_loop(self):
-        while True:
+        for i in itertools.count():
```
Comment (raylu) [inline futaba/delayed.py]: this feels like you'll run into issues once the number gets large. perhaps reset once it hits `delay_chunk_size` instead?

File: futaba/delayed.py [hunk 2]
```diff
@@ -40,20 +39,22 @@ def push(self, coro):
...
-
-    @property
-    def delay(self):
-        rate = self.queue.qsize() // self.bracket_size
-        return min(rate * self.rate, self.max_delay)
+            logger.debug("Got event, is #%d to be processed", i)
+            try:
+                await coro
+            except Exception as error:
+                logger.error("Error awaiting delayed event", exc_info=error)
```
Comment (raylu) [inline futaba/delayed.py]: before, exceptions caused this to bail. did you mean to continue instead of reraise?

File: futaba/delayed.py (hunk 1 above)
Comment (ammongit) [reply futaba/delayed.py]: Python is pretty good at supporting very large integers, I don't expect this to be a problem.

File: futaba/delayed.py (hunk 2 above)
Comment (ammongit) [reply futaba/delayed.py]: This doesn't reraise anything, it just prints the exception to the log so we can see it failed.

Comment (raylu) [reply futaba/delayed.py]: er, yeah, that's what I'm saying. before, exceptions stopped the loop. now, they don't. was that intentional?

Comment (ammongit) [reply futaba/delayed.py]: The loop should've never stopped, that was a bug. If the loop is dead then delayed events never get processed and just build up forever.

**Pull request merged**